src/DotVVM.Core/Controls/GridViewDataSet.cs

```diff
@@ -194,94 +293,44 @@ protected virtual void FillDataSet(GridViewDataSetLoadedData data)
-        public bool IsFirstPage { get => PagingOptions.IsFirstPage; }
-
-        /// <summary>
-        /// Determines whether the PageIndex represents the last page.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use PagingOptions.IsLastPage instead. This property will be removed in future versions.")]
-        public bool IsLastPage { get => PagingOptions.IsLastPage; }
-
-        /// <summary>
-        /// Calcualtes the total number of pages.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use PagingOptions.PagesCount instead. This property will be removed in future versions.")]
-        public int PagesCount { get => PagingOptions.PagesCount; }
-
-        /// <summary>
-        /// Calculates a list of page indexes for the pager controls.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use PagingOptions.NearPageIndexes instead. This property will be removed in future versions.")]
-        public IList<int> NearPageIndexes { get => PagingOptions.NearPageIndexes; }
-
-        /// <summary>
-        /// Gets or sets whether the sort order should be descending.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use SortingOptions.SortDescending instead. This property will be removed in future versions.")]
-        public bool SortDescending { get => SortingOptions.SortDescending; set => SortingOptions.SortDescending = value; }
-
-        /// <summary>
-        /// Gets or sets the name of the property that is used for sorting.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use SortingOptions.SortExpression instead. This property will be removed in future versions.")]
-        public string SortExpression { get => SortingOptions.SortExpression; set => SortingOptions.SortExpression = value; }
-
-
-        /// <summary>
-        /// Gets or sets the name of property that uniquely identifies the row - unique row ID, primary key etc.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use RowEditOptions.PrimaryKeyPropertyName instead. This property will be removed in future versions.")]
-        public string PrimaryKeyPropertyName { get => RowEditOptions.PrimaryKeyPropertyName; set => RowEditOptions.PrimaryKeyPropertyName = value; }
-
-        /// <summary>
-        /// Gets or sets the value of PrimaryKeyPropertyName property for the row that is currently edited.
-        /// </summary>
-        [Bind(Direction.None)]
-        [Obsolete("Use RowEditOptions.EditRowId instead. This property will be removed in future versions.")]
-        public object EditRowId { get => RowEditOptions.EditRowId; set => RowEditOptions.EditRowId = value; }
+        private async Task<GridViewDataSetLoadedData<T>> LoadData()
+        {
+            if (OnLoadingData != null && OnLoadingDataAsync != null)
+            {
+                //TODO: What type of exception should be here?
```

Comment:
Well that one just happens to be in DotVVM.Framework and therefore I can't reference it in DotVVM.Core without creating a circular dependency.

Maybe it wouldn't be such a bad idea to move DotvvmHttpException to DotVVM.Core.